x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/planner/QueryTranslator.java

```diff
@@ -68,9 +71,26 @@ public static Query toQuery(Expression e, TranslatorHandler handler) {
         for (ExpressionTranslator<?> translator : QUERY_TRANSLATORS) {
             translation = translator.translate(e, handler);
             if (translation != null) {
-                return translation;
+                break;
             }
         }
+        if (translation != null) {
+            if (translation instanceof ScriptQuery) {
+                // check the operators and the expressions involved in these operations so that all can be used
+                // in a doc-values multi-valued context
+                boolean multiValuedIncompatible = e.anyMatch(exp -> {
+                    if (exp instanceof Literal || exp instanceof FieldAttribute || exp instanceof Function) {
+                        return false;
+                    } else {
+                        return true;
+                    }});
```

Comment:
Potential collapsing:
```suggestion
                boolean multiValuedIncompatible = e.anyMatch(exp -> (exp instanceof Literal || exp instanceof FieldAttribute || exp instanceof Function) == false);
```